Sources/Boutique/Store.swift

```diff
@@ -65,16 +76,20 @@ public final class Store<Item: Codable & Equatable>: ObservableObject {
     public init(storage: StorageEngine, cacheIdentifier: KeyPath<Item, String>) {
         self.storageEngine = storage
         self.cacheIdentifier = cacheIdentifier
+        _ = self.setupTask // Start the lazy task in the background.
+    }
 
-        Task { @MainActor in
-            do {
-                let decoder = JSONDecoder()
-                self.items = try await self.storageEngine.readAllData()
-                    .map({ try decoder.decode(Item.self, from: $0) })
-            } catch {
-                self.items = []
-            }
-        }
+    /// Initializes a new ``Store`` for persisting items to a memory cache
+    /// and a storage engine, to act as a source of truth, and await for the ``items`` to load.
+    ///
+    /// - Parameters:
+    ///   - storage: A `StorageEngine` to initialize a ``Store`` instance with.
+    ///   - cacheIdentifier: A `KeyPath` from the `Item` pointing to a `String`, which the ``Store``
+    ///   will use to create a unique identifier for the item when it's saved.
+    @MainActor public init(storage: StorageEngine, cacheIdentifier: KeyPath<Item, String>) async throws {
```

Comment:
Nit/total personal preference: For functions I have a preference to putting annotations such as `@MainActor` on its own line above the function. For reference you can see that in the usage of `@_disfavoredOverload` in `Store.swift` as well.